src/main/java/net/minecrell/serverlistplus/core/replacement/DefaultPatternPlaceholder.java

```diff
@@ -91,14 +91,14 @@ public Object next() {
                     group = matcher.group(2);
                     int max = group == null ? DEFAULT_LIMIT : Integer.parseInt(group);
                     group = matcher.group(3);
-                    String delimiter = group == null ? DEFAULT_DELIMITER : group;
+                    String[] delimiters = ((group == null) ? DEFAULT_DELIMITER : group).split("(?<!\\\\)\\|");
```

Comment:
Can you pre-compile this pattern: add a constant (probably can't be static because of stupid weird inner classes) with `Pattern.compile("...")` then use `p.split(/* the string */)`